pkg/autoscaler/multiscaler.go

```diff
@@ -342,7 +342,7 @@ func (m *MultiScaler) tickScaler(ctx context.Context, scaler UniScaler, scaleCha
 
 		// Don't scale to zero if scale to zero is disabled.
 		if desiredScale == 0 && !m.dynConfig.Current().EnableScaleToZero {
```

Comment:
Since the KPA protects the scale-to-zero case, perhaps this check in the MultiScaler is not necessary at all?

Reply:
I removed this check before. Then I thought this could reduce kpa reconcile times. WDYT?

Reply:
Having this check in two places is more confusing than it brings us value through performance optimizations. In fact: Why are we reaching the scale bounds code with a desired scale of 0 at all? Shouldn't this be prevented by this check?

Reply:
Removed.

> In fact: Why are we reaching the scale bounds code with a desired scale of 0 at all? Shouldn't this be prevented by this check?

The KAP resync period is [30 seconds](https://github.com/knative/serving/blob/master/cmd/autoscaler/main.go#L85) at this point. KAP resync triggers `Reconcile` function and use -1 as desired scale value. `applyBounds` adjusts -1 to 0.